### DISCUSSION TITLE
QueryToListAsync to support the "IN" clause in Oracle SQL statements

### DISCUSSION THREAD

**Discussion by AppChenX** [linq2db/linq2db#4753] [Q&A] [answered]
How can I use QueryToListAsync to execute Oracle SQL statements with parameters that support "IN"? I couldn't find it in the test cases, or are there other methods?

**Comment by MaceWindu** [comment]
Could you be more specific what you want to achieve?

**Reply by AppChenX** [reply]
```
 public class TbTest{

   public string Id {get;set;}
   public string Code {get;set;}
   ..
}
//use flue mapping 

var dbContext=...

var sql="SELECT T.*  FROM TB_TEST T WHERE T.CODE IN  :P1";   //oracle

var lst=await dbContext..QueryToListAsync<TbTest>(sql, new { P1=new string[]{"A","B"} });
```
"I don't want to write the condition as WHERE CODE IN ('A', 'B')."  the  sql may be     such as Dapper .
SQL can be quite complex. The SQL I provided is a simple example, but I'm not sure how to write the parameters for the IN clause. I have tried many methods and none of them work.

**Reply by MaceWindu** [reply, marked as answer]
This is not supported currently https://github.com/linq2db/linq2db/issues/1645
